### PR TITLE
[Wallet] Add support for address pasting in send input field

### LINKED ISSUE #815: Send screen input should allow users to paste
### Expected Behavior

Users can paste and we show a little paste icon if we detect an address or phone number

### Current Behavior

Can't seem to paste, even when pressing and waiting, no paste window shown.

### PR DESCRIPTION
**Pull request by jmrossy**
### Description

- Add support for detecting addresses in a clipboard 
- Fix pasting into send field
- Improve modularity and typing of text input components 

### Tested

Loaded send with and without address in clipboard
Ran through send flow
Ran through invite flow

### Related issues

- Fixes #815 

### Backwards compatibility

Yes

wa-send-paste
wa-send-paste2

### CODE REVIEW

File: packages/mobile/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -10,6 +10,8 @@
   <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
   <uses-permission android:name="android.permission.CAMERA" />
 
+  <uses-sdk android:minSdkVersion="16" />
```
Comment (jmrossy) [inline packages/mobile/android/app/src/main/AndroidManifest.xml:13]: Adding this back, I mistakenly removed the line in my last PR but should have just removed the targetSDK field in it

File: packages/react-components/components/WithTextInputLabeling.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
+// HOC to add a label (e.g. an icon or a currency code) to a text input
```
Comment (jmrossy) [inline packages/react-components/components/WithTextInputLabeling.tsx:1]: FYI, this is mostly based on LabeledTextInput.tsx which was deleted

File: packages/react-components/components/WithTextInputPasteAware.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    onPressPate = async () => {
+      const clipboardContents = await Clipboard.getString()
+      this.props.onChangeText(clipboardContents)
+      this.setState({ isPasteIconVisible: false })
+    }
+
+    render() {
+      const { isPasteIconVisible } = this.state
+
+      // TODO(Rossy) Use a more paste-y instead of copy looking icon when we have one
```
Comment (jeanregisser) [inline packages/react-components/components/WithTextInputPasteAware.tsx:58]: Exactly what I thought when I saw the icon 😄

File: packages/react-components/components/WithTextInputPasteAware.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      try {
+        const clipboardContent = await Clipboard.getString()
+        if (
+          clipboardContent &&
+          clipboardContent !== this.props.value &&
+          this.props.shouldShowClipboard(clipboardContent)
+        ) {
+          this.setState({ isPasteIconVisible: true })
+        } else {
+          this.setState({ isPasteIconVisible: false })
```
Comment (jeanregisser) [inline packages/react-components/components/WithTextInputPasteAware.tsx:42]: nit: not really a problem here but this style of async usage could make it call `setState` after the component has been unmounted

**Review by jeanregisser — APPROVED** [review]
Looks good!

Though I understand the idea to separate the different TextInput features (paste and labeling), I'm a bit wary of the usage of HOC for this, I find it makes it non obvious to understand how to use the HOCs. 
What would happen if they aren't composed in the right order?
I might be missing something but having props to enable/disable labels and paste would seem simpler as a user of TextInput.

File: packages/react-components/components/WithTextInputPasteAware.tsx (hunk 4 above)
Comment (jmrossy) [reply packages/react-components/components/WithTextInputPasteAware.tsx:42]: Is there a general strategy on how to avoid that?

**Pull request merged**